Library/Homebrew/version.rb

```diff
@@ -387,6 +387,7 @@ def <=>(other)
     other = Version.new(other.to_s) if other.is_a? Integer
     return 1 if other.nil?
 
+    return 1 if other.respond_to?(:null?) && other.null?
     return unless other.is_a?(Version)
```

Comment:
The check had to go above the check that `other` is a version because `Version::NULL`'s anonymous class does not inherit from `Version`. (The type check feels like it's probably a smell, but out of scope).